libtiledbvcf/src/vcf/region.cc

```diff
@@ -154,15 +155,74 @@ void Region::parse_bed_file_htslib(
   if (regionsFile == nullptr)
     throw std::runtime_error("Error parsing BED file: " + bed_file_uri);
 
+  // If there is an index file process in parallel
+  if (regionsFile->tbx != nullptr) {
+    std::vector<std::future<std::list<Region>>> futures;
+    // Loop over the sequences in the index file and launch a task for each one
+    for (int i = 0; i < regionsFile->nseqs; i++) {
+      std::string chr = regionsFile->seq_names[i];
+      futures.emplace_back(
+          std::async(std::launch::async, [vfs, bed_file_uri, chr]() {
```

Comment:
Capture `vfs`, `bed_file_uri`, and `chr` by reference to avoid a copy. This should be safe since we wait on the tasks to complete before these variables are destructed.